File: leetcode/1281_subtract_the_product_and_sum_of_digits_of_an_integer.py
# -*- coding: utf-8 -*-

from math import prod
from typing import List


class Solution:
    def subtractProductAndSum(self, n: int) -> int:
        digits = self.getDigits(n)
        return prod(digits) - sum(digits)

    def getDigits(self, n: int) -> List[int]:
        digits = []
        while n:
            digits.append(n % 10)
            n //= 10
        return digits


if __name__ == '__main__':
    solution = Solution()

    assert 15 == solution.subtractProductAndSum(234)
    assert 21 == solution.subtractProductAndSum(4421)

File: leetcode/1281_subtract_the_product_and_sum_of_digits_of_an_integer.cpp
// Copyright (c) 2020 Jacopo Notarstefano

#include <cassert>

using namespace std;

class Solution {
 public:
    int subtractProductAndSum(int n) {
        int prod = 1, sum = 0;
        while (n) {
            prod *= n % 10;
            sum += n % 10;
            n /= 10;
        }
        return prod - sum;
    }
};

int main() {
    auto solution = Solution();

    assert(15 == solution.subtractProductAndSum(234));
    assert(21 == solution.subtractProductAndSum(4421));
}
